core/src/main/java/com/oath/oak/OakDetachedReadValueBufferSynced.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import java.util.ConcurrentModificationException;
+import java.util.function.Function;
+
+class OakDetachedReadValueBufferSynced extends OakDetachedReadValueBuffer {
+
+    private static final int MAX_RETRIES = 1024;
+
+    private final ValueUtils valueOperator;
+
+    /**
+     * In case of a search, this is the map we search in.
+     */
+    private final InternalOakMap<?, ?> internalOakMap;
+
+    OakDetachedReadValueBufferSynced(ValueUtils valueOperator, InternalOakMap<?, ?> internalOakMap) {
+        super(valueOperator.getHeaderSize());
+        this.valueOperator = valueOperator;
+        this.internalOakMap = internalOakMap;
+    }
+
+    @Override
+    protected <T> T transformBuffer(Function<OakAttachedReadBuffer, T> transformer) {
+        // Internal call. No input validation.
+
+        start();
+        try {
+            return transformer.apply(value);
+        } finally {
+            end();
+        }
+    }
+
+    private void start() {
+        // Use a "for" loop to ensure maximal retries.
```

Comment:
Why is buffer (of Slice) not read again? I think that I miss something here...

Reply:
We did not read the buffer again in the original value buffer as well. Nothing new here.